2007-02-15  Khem Raj  <kraj@xxxxxxxxxx>

       * sysdeps/unix/sysv/linux/i386/sysdep.h: Re-define __i686.
       * nptl/sysdeps/pthread/pt-initfini.c: Ditto.



Index: sysdeps/unix/sysv/linux/i386/sysdep.h
===================================================================
--- sysdeps/unix/sysv/linux/i386/sysdep.h	(revision 1469)
+++ sysdeps/unix/sysv/linux/i386/sysdep.h	(working copy)
@@ -29,6 +29,10 @@
 #include <dl-sysdep.h>
 #include <tls.h>
 
+#if defined __i686 && defined __ASSEMBLER__
+#undef __i686
+#define __i686 __i686
+#endif
 
 /* For Linux we can use the system call table in the header file
 	/usr/include/asm/unistd.h
Index: nptl/sysdeps/pthread/pt-initfini.c
===================================================================
--- nptl/sysdeps/pthread/pt-initfini.c	(revision 1469)
+++ nptl/sysdeps/pthread/pt-initfini.c	(working copy)
@@ -45,6 +45,11 @@
 /* Embed an #include to pull in the alignment and .end directives. */
 asm ("\n#include \"defs.h\"");
 
+asm ("\n#if defined __i686 && defined __ASSEMBLER__");
+asm ("\n#undef __i686");
+asm ("\n#define __i686 __i686");
+asm ("\n#endif");
+
 /* The initial common code ends here. */
 asm ("\n/*@HEADER_ENDS*/");
 
